crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -1424,27 +1449,7 @@ impl OlmMachine {
 
                 return Some(ProcessedToDeviceEvent::UnableToDecrypt(raw_event));
             }
-        };
-
-        // We ignore all to-device events from dehydrated devices - we should not
-        // receive any
-        match self.to_device_event_is_from_dehydrated_device(&decrypted, &e.sender).await {
-            Ok(true) => {
-                warn!(
-                    sender = ?e.sender,
-                    session = ?decrypted.session,
-                    "Received a to-device event from a dehydrated device. This is unexpected: ignoring event"
-                );
-                return None;
-            }
-            Ok(false) => {}
-            Err(err) => {
-                error!(
-                    error = ?err,
-                    "Couldn't check whether the event is from a dehydrated device",
-                );
-            }
-        }
+        }?;
```

Comment:
If you decide to apply my suggestion above:

```suggestion
        };
```